release: 2.68 (#15069)
canonical/snapd: NEWS.md

* FDE: update secboot to 30317622bbbc
* Snap components: make kernel components available on firstboot after either initramfs or ephemeral rootfs style install
* Snap components: mount drivers tree from initramfs so kernel modules are available in early boot stages
* Snap components: support remodeling to models that contain components
* Snap components: support offline remodeling to models that contain components
* Snap components: support creating new recovery systems with components
* Snap components: support downloading components with 'snap download' command
* Snap components: support sideloading asserted components
* AppArmor Prompting(experimental): improve version checks and handling of listener notification protocol for communication with kernel AppArmor
* AppArmor Prompting(experimental): make prompt replies idempotent, and have at most one rule for any given path pattern, with potentially mixed outcomes and lifespans
* AppArmor Prompting(experimental): timeout unresolved prompts after a period of client inactivity
* AppArmor Prompting(experimental): return an error if a patch request to the API would result in a rule without any permissions

> * AppArmor Prompting(experimental): warn of errors during snapd startup

Author: modified

> * AppArmor Prompting(experimental): support for downloading snap icons from store

Author: Agree, removed.

* AppArmor Prompting(experimental): do not log error when converting empty permissions to AppArmor permissions

> * Confdb(experimental): rename registries to confdbs

Author: I agree this looks somewhat weird, but its from the commit and it seems to be the agreed replacement for "registries" e.g.
![image](https://github.com/user-attachments/assets/9247f407-6bb8-4f54-92ce-a254c6528320)

> * Confdb(experimental): rename registries to confdbs

Reviewer: On the API level it was switched to confdb again https://github.com/canonical/snapd/pull/14994, cc: @miguelpires

> * Confdb(experimental): rename registries to confdbs

Reviewer 2: maybe we should have in parenthesis `(API /v2/registries => /v2/confdb)`

> * Confdb(experimental): rename registries to confdbs

Author: Changed to `Confdb(experimental): rename registries to confdbs (including API /v2/registries => /v2/confdb)`

> * Confdb(experimental): rename registries to confdbs

Reviewer 3: There are both of confdbs (experimental flag) and confdb (API) but yes, in this case, maybe the singular fits better

* Confidential VMs: snap-bootstrap support for loading partition information from a manifest file for cloudimg-rootfs mode
* Confidential VMs: snap-bootstrap: cloudimg-rootfs support for setting up the rootfs as an overlayfs with integrity protection
* dm-verity for essential snaps: add support for snap-integrity assertion
* Allow mksquashfs 'xattrs' when packing snap types os, core, base and snapd as part of work to support non-root snap-confine
* Upstream/downstream packaging changes and build updates
* Modify AppArmor template to allow owner read on @{PROC}/@{pid}/fdinfo/*
* Modify AppArmor template to allow using setpriv to run daemon as non-root user
* Improve error logs for malformed desktop files to also show which desktop file is at fault
* Provide more precise error message when overriding channels with grade during seed creation
* Expose 'snap prepare-image' validation parameter
* Add snap-seccomp 'dump' command that dumps the filter rules from a compiled profile
* Add fallback release info location /etc/initrd-release
* Added core-initrd to snapd repo and fixed issues with ubuntu-core-initramfs deb builds
* Remove robust-mount-namespace-updates experimental feature

> * Remove snapd-snap experimental feature

Author: robust-mount-namespace-updates experimental feature: looking at https://github.com/canonical/snapd/pull/10676 feature code was significantly changed, my interpretation is effectively removed due to a different implementation. So going to change to feature flag was removed.

snapd-snap experimental: here the feature and the flag was removed. making it more explicit.

* Changed snap-bootstrap to mount base directly on /sysroot
* Mount ubuntu-seed mounted as no-{suid,exec,dev}
* Mapping volumes to disks: add support for volume-assignments in gadget
* Fix silently broken binaries produced by distro patchelf 0.14.3 by using locally build patchelf 0.18
* Fix not refreshing due to outdated validation sets by fetching validation sets before checking for refresh candidates

> * Fix 'snap get' to produce compact listing for tty

Reviewer 4: ```suggestion
* Fix 'snap get' to produce compact listing for tty
* Fix progress reporting when stdout is on a tty, but stdin is not
```

Author: Thanks, yes this was added post release data collection and I forgot to add.
Added now.

* Provide more precise error message when overriding channels with grade during seed creation
* Expose 'snap prepare-image' validation parameter
* Add snap-seccomp 'dump' command that dumps the filter rules from a compiled profile
* Add fallback release info location /etc/initrd-release
* Added core-initrd to snapd repo and fixed issues with ubuntu-core-initramfs deb builds

> * Remove robust-mount-namespace-updates experimental feature flag

Reviewer: ```suggestion
* Remove stale robust-mount-namespace-updates experimental feature flag
```

Author: Thanks, changed

> * Remove snapd-snap experimental feature and feature flag

Reviewer 2: I think when we mention feature flag removal we should be clearer whether now they are fully by on default or they are simply ripped out. In the larger release notes on the forum we should make clear that having removing flags on will not have ill effects cc @ZeyadYasser

Reviewer: I need to dig, I removed it because the flag was stale and never used in the code base, I will try and figure out what happened for the feature itself.

Reviewer: The feature was removed, https://github.com/canonical/snapd/pull/7773 specifically https://github.com/canonical/snapd/pull/10676/commits/df6bbd5c64d2820836e74039726a10dfe844cd91 removes usage of the flag, and https://github.com/canonical/snapd/pull/10676/commits/2bba6a6c4278b45ff22064aef77e838df086634c removes the implementation.

For completeness, it was initially introduced in https://github.com/canonical/snapd/pull/7773.

Author: It is a bit open for interpretation, the original code is replaced by new code than improves the situation using a somewhat different approach. But its still in snap-update-ns/change.go.

In  my opinion, the closest to the truth is that the feature was accepted and the flag removed.

Author: In the `snapd-snap` experimental case, the feature was not accepted, and removed along with the flag that controls it.

Author: Perhaps we should standardize terminology a bit: "accepting/not accepting/rejecting" a feature

Author: Changed to:

```
Remove stale obust-mount-namespace-updates experimental feature flag
Remove snapd-snap experimental feature (rejected) and it's feature flag
```

> * Remove stale obust-mount-namespace-updates experimental feature flag

Reviewer: ```suggestion
* Remove stale robust-mount-namespace-updates experimental feature flag
```

Author: oh no, and thanks

Author: fixed

* Changed snap-bootstrap to mount base directly on /sysroot
* Mount ubuntu-seed mounted as no-{suid,exec,dev}
* Mapping volumes to disks: add support for volume-assignments in gadget
* Fix silently broken binaries produced by distro patchelf 0.14.3 by using locally build patchelf 0.18

> * Fix not refreshing due to outdated validation sets by fetching validation sets before checking for refresh candidates

Reviewer 2: this one sounds strange as I thought it was about listing candindate refreshes, not refreshes in general cc @andrewphelpsj @miguelpires

Reviewer 3: It is, if it's referring to this fix https://github.com/canonical/snapd/pull/14907
maybe :
```suggestion
* Fix mismatch between listed refresh candidates and actual refresh due to outdated validation sets
```

Author: @pedronis, you are right, it addresses `snap refresh --list` behavior, not refreshes itself.

Looking at the code, I have question. 

It seems like validation sets can be updated by account holder to affect any snaps at any time by creating new validation set asserts. 

We check that there are existing validation sets, if not, we do not refresh but exit. What am I missing here? @pedronis @miguelpires

Reviewer 2: I'm not sure I understand the question,  a device only tracks a specific set of validation-sets set by the model or via the API

Author: Thanks @miguelpires, used your suggestion

* fwupd interface: allow access to shim and fallback shim
* mount-control interface: add mount option validator to detect mount option conflicts early
* cpu-control interface: add read access to /sys/kernel/irq/<IRQ>
* locale-control interface: changed to be implicit on Ubuntu Core Desktop
* microstack-support interface: support for utilizing of AMD SEV capabilities
* u2f interface: added missing OneSpan device product IDs
* auditd-support interface: grant seccomp setpriority

> * opengl interface: enable parsing of nvidia driver information files

Author: Proposed this style 
```
* Interfaces: fwupd | allow access to dell bios recovery
* Interfaces: fwupd | allow access to shim and fallback shim
```
